CBMC: Proofs for native Armv8.4-A Keccak variants failing on x86

Currently, the following 3 CBMC proofs fail on x86: 
- keccak_f1600_x1_native_aarch64_v84a
- keccak_f1600_x4_native_aarch64_v84a
- keccak_f1600_x4_native_aarch64_v8a_v84a_scalar_hybrid

The error is: 
```
/tmp/mlkem-native/mlkem/src/fips202/native/aarch64/x4_v8a_v84a_scalar.h: In function 'mlk_keccak_f1600_x4_native':

/tmp/mlkem-native/mlkem/src/fips202/native/aarch64/x4_v8a_v84a_scalar.h:23:1: error: failed to find symbol 'MLK_SYS_CAP_SHA3'

   if (!mlk_sys_check_capability(MLK_SYS_CAP_SHA3))

CONVERSION ERROR
```

This symbol is only defined on an AArch64 system.

--- a/mlkem/src/sys.h
+++ b/mlkem/src/sys.h
@@ -217,15 +217,10 @@
 /* System capability enumeration */
 typedef enum
 {
-#if defined(MLK_SYS_X86_64)
-  MLK_SYS_CAP_AVX2
-#elif defined(MLK_SYS_AARCH64)
+  /* x86_64 */
+  MLK_SYS_CAP_AVX2,
+  /* AArch64 */
   MLK_SYS_CAP_SHA3
-#else
-  /* C90 does not allow empty enums, so use a dummy value
-   * for architectures other than AArch64 and x86_64. */
-  MLK_SYS_CAP_DUMMY
-#endif
 } mlk_sys_cap;
 
 #if !defined(MLK_CONFIG_CUSTOM_CAPABILITY_FUNC)

